error.go

```diff
@@ -0,0 +1,185 @@
+package slogx
+
+import (
+	"errors"
+	"log/slog"
+)
+
+const KeyBadKey = "!BADKEY"
+
+var cfg = &config{} //nolint:gochecknoglobals // By design.
+
+type errorAttrs struct { //nolint:errname // Custom naming.
+	err   error
+	attrs []slog.Attr
+}
+
+type config struct {
+	groupTopErrorAttrs  bool
+	inlineSubErrorAttrs bool
+}
+
+// Error returns string value of errorAttrs error.
+func (e errorAttrs) Error() string { return e.err.Error() }
+
+// Unwrap returns errorAttrs error.
+func (e errorAttrs) Unwrap() error { return e.err }
+
+type errorAttrsOption func()
+
+// NewError returns errorAttrs error that contains given err and args,
+// modified to []slog.Attr.
+func NewError(err error, args ...any) error {
+	if err == nil {
+		return nil
+	}
+	e := errorAttrs{err: err}
+	if len(args) > 0 {
+		e.attrs = argsToAttrSlice(args)
+	}
+	return e
+}
+
+// NewErrorAttrs returns errorAttrs error that contains given err and attrs.
+func NewErrorAttrs(err error, attrs ...slog.Attr) error {
+	if err == nil {
+		return nil
+	}
+	e := errorAttrs{err: err}
+	if len(attrs) > 0 {
+		e.attrs = attrs
+	}
+	return e
+}
+
+type errorNoAttrs struct { //nolint:errname // Custom naming.
+	err error
+}
+
+// Error returns string value of errorNoAttrs error.
+func (e errorNoAttrs) Error() string { return e.err.Error() }
+
+// Unwrap returns errorNoAttrs error.
+func (e errorNoAttrs) Unwrap() error { return e.err }
```

Comment:
Can't figure out how to test this method.
It is not used in code.
This is not export type.
`NewErrorNoAttrs` returns `error` type, that doesn't have method `Unwrap`, so I can't call it in tests.